### PR TITLE
3017.5 fix clippy warnings

### PR DESCRIPTION
**Pull request by roxy-dao**
Part of #3017 

# 👩🏻‍💻 What does this PR do? 
More clippy fixes 😆 

# 🧪 How has/should this change been tested? 
- [ ] Make sure tests still run and nothing is borken 😄

### CODE REVIEW

**Comment by clemens-msupply** [comment]
There are some merge conflicts...

**Comment by roxy-dao** [comment]
> There are some merge conflicts...

Yup, seems like `returns feature` got merge. Will fix soon.

File: server/repository/src/db_diesel/assets/asset_internal_location_row.rs [hunk 1]
```diff
@@ -117,7 +117,7 @@ impl<'a> AssetInternalLocationRowRepository<'a> {
 
 impl Upsert for AssetInternalLocationRow {
     fn upsert_sync(&self, con: &StorageConnection) -> Result<(), RepositoryError> {
-        let _change_log_id = AssetInternalLocationRowRepository::new(con).upsert_one(self)?;
```
Comment (jmbrunskill) [inline server/repository/src/db_diesel/assets/asset_internal_location_row.rs:120]: Personally I prefer this version, as it clearly identifies what we're throwing away.

File: server/repository/src/db_diesel/changelog/changelog.rs [hunk 2]
```diff
@@ -430,18 +430,12 @@ fn create_filtered_outgoing_sync_query(
     // Central Records
 
     let central_sync_table_names: Vec<ChangelogTableName> = ChangelogTableName::iter()
-        .filter(|table| match table.sync_style() {
-            ChangeLogSyncStyle::Central => true,
-            _ => false,
-        })
+        .filter(|table| matches!(table.sync_style(), ChangeLogSyncStyle::Central))
```
Comment (jmbrunskill) [inline server/repository/src/db_diesel/changelog/changelog.rs:433]: Interesting, haven't seen this macro before...

File: server/service/src/processors/transfer/shipment/test.rs [hunk 3]
```diff
@@ -578,7 +578,7 @@ impl ShipmentTransferTester {
         );
         assert_eq!(inbound_shipment.colour, None);
         assert_eq!(inbound_shipment.user_id, None);
-        assert_eq!(inbound_shipment.on_hold, false);
+        assert!(!inbound_shipment.on_hold);
```
Comment (jmbrunskill) [inline server/service/src/processors/transfer/shipment/test.rs:581]: There's quite a few of these changes. I'm personally find the old version easier to read but I'm ok with this if we want it, also happy to disable this lint...

**Review by jmbrunskill — APPROVED** [review]
I'd personally prefer to have 
`assert_eq!(parsed_valid_response.under_temperature, false);`
rather than
`assert!(!parsed_valid_response.under_temperature);`

I think it's partly about the `!(!` being so many `!`'s so close to each other I find it a bit hard to read.

If you agree I'd suggest we disable that lint?

File: server/service/src/processors/transfer/shipment/test.rs (hunk 3 above)
Comment (roxy-dao) [reply server/service/src/processors/transfer/shipment/test.rs:581]: Both same readability to me... can ask team~

Comment (clemens-msupply) [reply server/service/src/processors/transfer/shipment/test.rs]: Find the first version better as well but I am ok going with the clippy suggestion

**Pull request merged**